packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -264,6 +264,21 @@ def draw_markers(self, gc, marker_path, marker_trans, path,
         super().draw_markers(gc, marker_path, marker_trans, path,
                              trans, rgbFace)
 
+    def draw_image(self, gc, x, y, im, transform=None):
+        im = np.flipud(im)
+        h, w, d = im.shape
+        im = np.ravel(np.uint8(np.reshape(im, (h * w * d, -1)))).tobytes()
+        img_data = ImageData.new(im, w, h)
+        self.ctx.save()
+        in_memory_canvas = document.createElement('canvas')
+        in_memory_canvas.width = self.ctx.width
+        in_memory_canvas.height = self.ctx.height
+        in_memory_canvas_context = in_memory_canvas.getContext('2d')
+        in_memory_canvas_context.putImageData(img_data, x,
+                                              self.ctx.height - y - h)
```

Comment:
This creates an in_memory_canvas that is the full size of the plot.  It would be more efficient to make it the size of the incoming image, and then offset it into its final destination in the call to drawImage below.

Reply:
Done!